### PR TITLE
WIP: mingw-w64 support

### PR DESCRIPTION
**Pull request by PureTryOut**
I spent way too much time on this already. I finally got it through the CMake configuring phase but it now fails to build on a simple include... I don't know enough about what's going on and I'm mainly posting this hoping someone else can take over.

### CODE REVIEW

**Comment by janoc** [comment]
That seems to fail on:
```
D:/a/_temp/msys64/mingw64/x86_64-w64-mingw32/include/stddef.h:18:10: error: expected ‘;’ before ‘extern’
   18 |   _CRTIMP extern int *__cdecl _errno(void);
      |          ^~~~~~~
      |          ;`
```

_CRTIMP is not defined somewhere? That sounds like some windows-specific switches need to be passed to the compiler :(

File: CMakeLists.txt [hunk 1]
```diff
@@ -53,6 +53,8 @@ if(WIN32)
 	if(MSVC)
 		# Minimum requirement is WinXP
 		add_definitions(-D_WIN32_WINNT=0x0501)
+	else()
+		add_definitions(-D_WIN32)
```
Comment (jcelerier) [inline CMakeLists.txt:57]: not necessary, it's already defined by mingw-w64-gcc

File: cmake/FindWinHID.cmake [hunk 2]
```diff
@@ -219,6 +221,7 @@ else()
 		${MINGWSEARCH_INCLUDE_DIRS}
 		/mingw
 		PATH_SUFFIXES
+		x86_64-w64-mingw32/include
```
Comment (jcelerier) [inline cmake/FindWinHID.cmake:224]: why is that needed ? here with mingw-w64 it works fine as-is

File: CMakeLists.txt (hunk 1 above)
Comment (janoc) [reply CMakeLists.txt:57]: Better safe than sorry - that code is catch-all clause that covers all Windows compilers except MSVC. I.e. what happens if someone tries the Intel compiler with it?

File: cmake/FindWinHID.cmake (hunk 2 above)
Comment (janoc) [reply cmake/FindWinHID.cmake:224]: I suspect it is a leftover from an older version of MinGW? I would also expect that the compiler should add its own include directory automatically.

File: CMakeLists.txt (hunk 1 above)
Comment (jcelerier) [reply CMakeLists.txt:57]: well, ICC targetting windows will define _WIN32. But for instance when under cygwin's GCC, it *should not* be defined because even if it produces a windows .exe, it is in all other aspects a reimplemented Unix which provides the Unix API and not the Windows one (e.g it defines `__unix__`, not `_WIN32`) as it was made to port from unices easily. MinGW-w64's compilers on the other hand are GCC and clang configured to target the Win32 API directly and as such define _WIN32 and not `__unix__`

Comment (janoc) [reply CMakeLists.txt:57]: OK and does building with Cygwin actually define the WIN32 variable that is tested before it gets to this point? If it does, then you are possibly right but I would hesitate to add Cygwin specific stuff.

File: cmake/FindWinHID.cmake (hunk 2 above)
Comment (janoc) [reply cmake/FindWinHID.cmake:224]: This could be removed, IMO - feel free to remove the line in your PR if it works without it.

**Comment by janoc** [comment]
Resolved in #131
